packages/components/src/Combobox/components/ComboboxAction/ComboboxAction.tsx

```diff
@@ -1,24 +1,39 @@
-import React from "react";
+import React, { useContext } from "react";
 import styles from "./ComboboxAction.css";
 import { Typography } from "../../../Typography";
 import { ComboboxActionProps } from "../../Combobox.types";
+import { ComboboxContext } from "../../ComboboxProvider";
+
+export function ComboboxAction(props: ComboboxActionProps) {
+  const { searchValue } = useContext(ComboboxContext);
+
+  if (props.visible) {
+    const isVisible =
+      typeof props.visible === "function" && !props.visible({ searchValue });
+
+    if (isVisible || !props.visible) {
+      return null;
+    }
+  }
+
+  const options = { searchValue };
+  const computedLabel =
+    typeof props.label === "string" ? props.label : props.label(options);
 
-export function ComboboxAction(props: ComboboxActionProps): JSX.Element {
   return (
     <div className={styles.actionContainer}>
       <button
         className={styles.actionButton}
-        onClick={props.onClick}
+        onClick={e => props.onClick(e, options)}
         type="button"
-        aria-label={props.label}
       >
         <Typography
           element="span"
           size="base"
           textColor="green"
```

Comment:
```suggestion
          textColor="interactive"
```

😉

Reply:
It actually has to be `green`? `interactive` is on mobile Typography, not web @chris-at-jobber

Reply:
🤯 OK separate task... "Allow Typography to consume semantic colors" 😄